Fixed for tables with deleted rows and 9.3 version support

sainsb opened the pull request:
So... the library wasn't parsing some filegdbs for me, here is an example (sorry, it's big)
http://www.fs.fed.us/r6/data-library/gis/okanogan/documents/Transportation.zip

I discovered that the number of records in the tablex function would sometimes return incorrect numbers (deleted rows(?)); to get the real number of rows in a table, you have to get it from the gdbtable.

Here is an example from Evan:
tablex number of rows:
https://github.com/rouault/dump_gdbtable/blob/master/dump_gdbtable.py#L171

table number of rows:
https://github.com/rouault/dump_gdbtable/blob/master/dump_gdbtable.py#L179

The code in this PR passes the gdbtable buffer into the rows function and then gets the numrows, but it could also parse the numrows first and pass that int val in...however you want to implement...

sainsb committed:
Fixed for tables with deleted rows

calvinmetcalf commented on lib/tablex.js:
```diff
@@ -1,13 +1,14 @@
-  var data = new DataView(buffer);
-  var offset = 8;
+module.exports = function (buffer, bufferx){
+  var tbl_num_rows = (new DataView(buffer)).getUint32(4,true);
```
Style nit, camelCase variable name, space after the coma and I belive the parentheses are Un needed due to the new operator having higher precedence the the dot operator.

calvinmetcalf commented on lib/tablex.js:
```diff
@@ -1,13 +1,14 @@
-    rows[i++] = data.getUint32(offset, true);
+  while(i<tbl_num_rows){
+    var val = data.getUint32(offset,true);
+    if(val==0){offset += 5;continue;}
```
Triple equals please

calvinmetcalf replied:
This would probably be better written as if (val ! == 0) { row[i++... 

Because the offset gets increased either way.

calvinmetcalf commented:
A few formating issues which are no big deal if you can't fix it can fix it when I merge. 

Which will be sometime next week as I'm at the airport about to go on vacation. 

(a test would be nice too if you are able to get them working)

calvinmetcalf commented:
But thank you this looks great.

sainsb committed:
Update fromZip.js to include fgdb version num

From Rouault:
For FileGDB v10, the first 8 (a00000001 to a00000008) files seem to be reserved for database information
For FileGDB v9, the first 36 (a00000001 to a00000024) files seem to be reserved for database information 

This fix will accomodate both generations of the fgdb.

sainsb committed:
Style nits, camelcase, tripleequals, better writed

sainsb committed:
Fixed some spacing and some camel casing

sainsb commented:
I think I've messed this up...sorry. I made the fixes you suggested and also another fix for fgdb version. I probably should've created a branch for the version fix... my bad.
